### ISSUE TITLE
Using a 200 Rewrite to a headless CMS results in 504

### ISSUE THREAD

**Issue by SethGunnells** [aws-amplify/amplify-console#390]
** Please describe which feature you have a question about? **
This question is related to 200 rewrites.

** Provide additional details**
I have a headless CMS (Craft CMS) that I'm hosting on an EC2 in the same region and account as my Amplify App. I've set up a 200 rewrite like so:
```
Source: /admin/<*>
Target: http://somedomain.com/admin
Redirect Type: rewrite (200)
```

When I attempt to go to `myapp.com/admin`, the page sits in a loading state for maybe 10 seconds and then cuts to a CloudFront 504 page. I'm not sure what to make of it. A 301 or 302 redirect works so worst case scenario I could create a subdomain, point it to my CMS, and redirect from /admin to the CMS control panel URL, but ideally I'd like the URL not to change at all.

**Comment by kahdojay** [comment]
@SethGunnells thanks for reporting this, we'll have to take a closer look to see whether this is a bug on our end. Are you able to use the redirect workaround for the time being?

**Comment by SethGunnells** [comment]
The redirect is working fine for now. It may be the only viable solution anyway, but I figured giving a rewrite a chance would be worthwhile.

**Comment by bogdandanila** [comment]
I sort of had the same issue happen to me. I set it up about a week ago and found it broken today. Here's my experience:

I have set up a 200 redirect on my Amplify front-end app to point to an API Gateway like this:
```
{
        "source": "/api/<*>",
        "target": "https://sg4uybb3v3.execute-api.us-east-2.amazonaws.com/stg/api/<*>",
        "status": "200",
        "condition": null
},
```
But I kept getting a 302 redirect response, even though calling the gateway directly returned the correct json respone.

After many attempts, I finally added a new redirect from `"source": "/proxy/<*>"` to the same gateway and changed the URLs in my front-end app. Everything works fine now, but calling /api/* still returns 302.

Hope this helps!
